packages/gatsby/src/utils/babel-loader.js

```diff
@@ -0,0 +1,86 @@
+const babelLoader = require(`babel-loader`)
+
+module.exports = ({ stage, browserslist, resolve = require.resolve }) =>
+  babelLoader.customLoader(babel => {
+    const presetEnv = babel.createConfigItem([
+      resolve(`@babel/preset-env`),
+      {
+        loose: true,
+        modules: false,
+        useBuiltIns: `usage`,
+        targets:
+          stage === `build-html`
+            ? { node: `current` }
+            : { browsers: browserslist },
+      },
+    ])
+    const presetReact = babel.createConfigItem([
+      resolve(`@babel/preset-react`),
+      {
+        useBuiltIns: true,
+        pragma: `React.createElement`,
+        development: stage === `develop`,
+      },
+    ])
+    const presetFlow = babel.createConfigItem(resolve(`@babel/preset-flow`))
+    const classProperties = babel.createConfigItem(
+      resolve(`@babel/plugin-proposal-class-properties`)
+    )
+    const dynamicImport = babel.createConfigItem(
+      resolve(`@babel/plugin-syntax-dynamic-import`)
+    )
+    const dynamicImportNode = babel.createConfigItem(
+      resolve(`babel-plugin-dynamic-import-node`)
+    )
+    const runtime = babel.createConfigItem(
+      resolve(`@babel/plugin-transform-runtime`)
+    )
+    const hotLoader = babel.createConfigItem(resolve(`react-hot-loader/babel`))
+    const graphqlQueries = babel.createConfigItem(
+      resolve(`babel-plugin-remove-graphql-queries`)
+    )
+
+    return {
+      // Passed the loader options.
+      customOptions(options) {
+        return {
+          loader: {
+            cacheDirectory: true,
+            babelrc: false,
+            sourceType: `unambiguous`,
+            ...options,
+          },
+        }
+      },
+
+      // Passed Babel's 'PartialConfig' object.
+      config(partialConfig) {
+        // TODO: we should run partialConfig through the gatsby babel API plugin
+        // The problem there is this is sync and the API run is async :/
```

Comment:
this is an interesting problem. we have no way of doing sync node api calls...

Reply:
I guess we could just do them all ahead of time?